pages/home.js

```diff
@@ -0,0 +1,325 @@
+        const life = dogBreeds[0].map((a) => a.life_span);
+        const temperament = dogBreeds[0].map((a) => a.temperament);
+        const dogBreedGroup = dogBreeds[0].map((a) => a.breed_group);
+
+        if (!dogName) {
+          return 'Oh there is no information, enjoy the image';
+        } else {
+          setDogImageUrl(newUrl);
+          setName(dogName);
+          setLifeSpan(life);
+          setChar(temperament);
+          setBreedGroup(dogBreedGroup);
+        }
+
+        //console.log(result.map((dog) => dog.breeds));
+      })
+      .then((error) => {
+        return error;
+      });
+  };
+
+  //const images = [];
+
+  //  function preload() {
+  //    // use process.browser => window/Image not defined
+  //    if (process.browser) {
+  //      for (let i = 0; i < images.length; i++) {
+  //        images[i] = new window.Image();
+  //        images[i].src = preload.images[i];
+  //      }
+  //      setPreloadImg('/loading1.jpg');
+  //    }
+  //  }
+
+  function changeImage(e) {
+    e.preventDefault();
+    fetchData();
+  }
+
+  // ---------------- image slide ----------------------------
+  const pics = [
+    '/about-us-dog.jpg',
+    '/favicon.jpg',
+    '/bullterrier.jpg',
+    '/englischedogge.jpg',
+    
+  ];
+  const indexStart = 0;
+  const [index, setIndex] = useState(indexStart);
+  const [move, setMove] = useState(false);
+  const [next, setNext] = useState();
+
+  // Follow the error and use callback wrap
+  const getNextIndex = useCallback(
+    (idx) => {
+      return idx >= pics.length - 1 ? 0 : index + 1;
+    },
+    [index, pics.length],
+  );
+
+  const setIndexes = useCallback(
+    (idx) => {
+      setIndex(idx);
+      setNext(getNextIndex(idx));
+    },
+    [getNextIndex],
+  );
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setMove(true);
+
+      setTimeout(() => {
+        setMove(false);
+        setIndexes(getNextIndex(index));
+      });
+    }, 4000);
+    // Here clear interval, although images will go faster and crazy !
+    return () => clearInterval(interval);
+    // Here I have to wrap dependensies inside Array
+  }, [index, getNextIndex, setIndexes]);
+
+  // -------------------  Preload image ------------------------
+  //  function preload(url, callback) {
+  //    if (process.browser) {
+  //      let image = new window.Image();
+  //      if (image.complete) {
+  //        callback.call(image);
+  //        return;
+  //      }
+  //      image.onload = () => {
+  //        callback.call(image);
+  //      };
+  //      image.src = url;
+  //    }
+  //  }
+  //  preload('/loading1.jpg', () => {
+  //    console.log('IMAGE LOADED');
+  //  });
+
+  function goSum() {
+    const sumDogs = {
+      dogImageUrl: dogImageUrl,
+      name: name,
+      lifeSpan: lifeSpan,
+      char: char,
+      breedGroup: breedGroup,
+    };
+
+    const favorite = Cookies.getJSON('sum') || [];
+    favorite.push(sumDogs);
+    Cookies.set('sum', favorite);
```

Comment:
I would suggest moving this information to the database, instead of saving it in a cookie.

To make sure that the user can only update their own information, you can add a `users` table and Registration + Login.